src/outpack/filestore.py

```diff
@@ -2,39 +2,44 @@
 import os.path
 import shutil
 import stat
+import tempfile
+from pathlib import Path
 
 from outpack.hash import Hash, hash_parse, hash_validate_file
 
 
 class FileStore:
     def __init__(self, path):
-        self._path = path
+        self._path = Path(path)
         os.makedirs(path, exist_ok=True)
 
     def filename(self, hash):
         dat = hash_parse(hash)
-        return os.path.join(
-            self._path, dat.algorithm, dat.value[:2], dat.value[2:]
-        )
+        return self._path / dat.algorithm / dat.value[:2] / dat.value[2:]
 
-    def get(self, hash, dst):
+    def get(self, hash, dst, overwrite):
```

Comment:
```suggestion
    def get(self, hash, dst, *, overwrite):
```

Generally nice for booleans to be kw arguments only, as it makes the call site explicit. Could give it a default too.